ShacShifter/JekyllRDFSerializer.py

```diff
@@ -0,0 +1,133 @@
+import logging
+import uuid
+
+# example class for Jekyll-RDF
+class JekyllRDFSerializer:
+    """A Serializer that writes Template for JekyllRDF"""
+
+    logger = logging.getLogger('ShacShifter.JekyllRDFSerializer')
+    content = []
+    outputfile = ''
+
+    def __init__(self, nodeShapes, outputfile):
+        try:
+            fp = open(outputfile, 'w')
+            self.outputfile = outputfile
+            fp.close
+        except Exception:
+            raise Exception('Can''t write to file {}'.format(outputfile))
+
+        self.content.append('''<html><head>
+                                <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta.2/css/bootstrap.min.css">
+                            </head><body>\n''')
+        self.logger.debug(nodeShapes)
+        self.content.append('<div class ="container">\n')
+        self.content.append("<h2>{{ page.rdf.iri }}</h2>\n")
+        self.content.append("""{% assign type = page.rdf | rdf_property: "<http://www.w3.org/1999/02/22-rdf-syntax-ns#type>" %}\n""")
+
+        for nodeShape in nodeShapes:
+            self.nodeShapeEvaluation(nodeShapes[nodeShape], fp)
+        self.content.append("""</div><div>
+                    <hr>
+                    Footer<br/>
+                    The data ist taken from: diggr.project
+                </div>""")
+        self.content.append('</body></html>')
+        self.saveToFile()
+
+    def saveToFile(self):
+        fp = open(self.outputfile, 'w')
+        fp.write(''.join(self.content))
+        fp.close
+
+    def nodeShapeEvaluation(self, nodeShape, fp):
+        """Evaluate a nodeShape.
+
+        args:   nodeShape a nodeShape object
+                fp
+        """
+
+        if len(nodeShape.targetClass) > 1:
+            self.content.append("Uebersicht")
+
+            for tClass in nodeShape.targetClass:
+                self.content.append(
+                    '<h2>{type}</h2><br><h3>{short}</h3><br>'.format(
+                        type=tClass, short=tClass.rsplit('/', 1)[-1]))
+
+        elif len(nodeShape.targetClass) == 1:
+            self.content.append('{{% if type.iri == "{type}" %}}\n\n'.format(
+                type=nodeShape.targetClass[0].lower()))
+            self.content.append("Ressource: <strong>{resource}</strong> ({type})<br><br>\n\n".format(
+                resource=nodeShape.targetClass[0].rsplit('/', 1)[-1], type=nodeShape.targetClass[0]))
+
+
+        for nodes in nodeShape.targetNode:
+            self.logger.debug(nodes)
+        self.logger.debug(
+            'The following ressources need to be Objects of those predicates'
+            )
+
+        for nodes in nodeShape.targetObjectsOf:
+            self.logger.debug(nodes)
+
+        for nodes in nodeShape.targetSubjectsOf:
+            self.logger.debug(nodes)
+
+        for property in nodeShape.properties:
```

Comment:
Starting here, until line 88 are calls to nodeShape.targetClass without prior testing if targetClass has any entries. There should be a more viable way to get the shapeName, Especially if no targetClass exists (checkable with nodeShape.isSet[variableName] too)